endpoints/horses/index.js

```diff
@@ -0,0 +1,190 @@
+var iconv = require('iconv-lite');
+var $ = require('cheerio');
+var h = require('apis-helpers');
+var app = require('../../server');
+
+function query_data(id, name, origin, microchip, callback) {
+  var url = 'http://www.worldfengur.com/freezone_horse.jsp?c=EN';
+  var headers = {
+    'User-Agent': h.browser(),
+    'Accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8',
+    'Content-Type': 'application/x-www-form-urlencoded'
+  };
+
+  // Encoding parameters so Icelandic characters will work
+  // Hacking around to get ISO 8859-1 encoded string, pls unhack if you know better way
+  var encoded_name = encodeURIComponent(escape(name)).replace(/%25/g,"%");
+  var encoded_origin = encodeURIComponent(escape(origin)).replace(/%25/g,"%");
+  var form_data = (
+    'fnr=' + id +
+    '&nafn=' + encoded_name +
+    '&uppruni=' + encoded_origin +
+    '&ormerki=' + microchip +
+    '&leitahnappur=Search+&leita=1'
+  );
+
+  request.post({
+    headers: headers,
+    url: url,
+    body: form_data,
+    encoding: null
+  }, function(error, response, body) {
+    if(error || response.statusCode !== 200) {
+      callback(error, response, '');
+    }
+    var html_page = iconv.decode(body, 'iso-8859-1');
+    callback(error, response, html_page);
+  });
+}
+
+function parse_data(html_page) {
+  var data = $(html_page);
+  // td_elements:
+  // is a list of text strings from all td elements in the returned html_page
+  // we find data from the list by searching for key labels
+  // we assume the next value in the list after a key label is a corresponding value
+  var td_elements = [];
+  data.find('div.complete div table').each(function() {
+    $(this).find('td').each(function() {
+      td_elements.push($(this).text());
+    });
+  });
+  // params:
+  // name is the name of a single param
+  // label is the label used on the webpage for a given param
+  // value is the value of a given param, and null if it wasn't found in td_elements
+  var params = [
+    {name: 'id',              label: 'FEIF ID',                     value: null},
+    {name: 'name_and_origin', label: 'Name and origin',             value: null},
+    {name: 'ueln',            label: 'UELN',                        value: null},
+    {name: 'date_of_birth',   label: 'Date of birth',               value: null},
+    {name: 'color_code',      label: 'Colour code',                 value: null},
+    {name: 'color',           label: 'Colour ',                     value: null},
+    {name: 'country_located', label: 'Country of current location', value: null},
+    {name: 'fate',            label: 'Fate ',                       value: null},
+    {name: 'microchip',       label: 'Microchip',                   value: null},
+    {name: 'father',          label: 'Sire',                        value: null},
+    {name: 'mother',          label: 'Dam',                         value: null},
+  ];
+  var labels = params.map(function(x){return x['label'];});
```

Comment:
In recent versions of node we have [arrow functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions) so we can now write anonymous functions like this:

```js
var labels = params.map(x => x.label);
```

which is a little bit cleaner, but don't bother changing this now unless you really want. I'm working on getting our code linted and using this syntax in the `eslint` branch. (You can check it's progress in #274 if you are interested).